### PR TITLE
'Prepare' the HTTP response based on the request before it is sent

### PR DESCRIPTION
**Pull request by orottier**
see #449

I'm pretty sure this is not the right place to perform the response preparation, but could not come up with a better place.

### CODE REVIEW

File: src/Http/ResponseFactory.php [hunk 1]
```diff
@@ -20,7 +20,9 @@ class ResponseFactory
      */
     public function make($content = '', $status = 200, array $headers = [])
     {
-        return new Response($content, $status, $headers);
+        $response = new Response($content, $status, $headers);
+
+        return $response->prepare(app()->request);
```
Comment (GrahamCampbell) [inline src/Http/ResponseFactory.php]: This is not the correct way to resolve the current request.

**Comment by GrahamCampbell** [comment]
I think not preparing was intentionally for performance.

**Comment by orottier** [comment]
Interesting.. Not preparing the response means you are stuck on HTTP 1.0 if I read it right
https://github.com/symfony/http-foundation/blob/master/Response.php#L305

Surely you don't want to sacrifise HTTP 1.1 features for this performance optimization?

**Comment by GrahamCampbell** [comment]
We could just force a higher version tbh.

**Comment by orottier** [comment]
Well, actually I don't really understand why the framework is juggling around with HTTP versions. I guess it is a construction to set the HTTP status code and reason phrase in a flexible way.

`prepare` does not look like a very intensive operation btw, just playing around with the headers bag. I could override with a more minimal approach if you wish. So we have some HTTP/1.0 support left

**Comment by GrahamCampbell** [comment]
The intensive operation is building a request object. We don't bother with that normally.

**Comment by orottier** [comment]
I've forced all responses to be sent over HTTP/1.1, which solves the Keep-Alive issues
Please let me know if that fits Lumen requirements!

**Comment by taylorotwell** [comment]
I would suggest using a middleware for this purpose.

**Comment by orottier** [comment]
Hi @taylorotwell, do you mean in the framework or the application code?

**Pull request closed without merging**